backend/src/routes/api/profileRoutes.js

```diff
@@ -34,4 +37,22 @@ const getUserProfile = async (req, res, next) => {
   }
 };
 
+async function getUsersMedications(req, res, next) {
+  try {
+    const currentProfile = await Profile.findById(req.params.id);
+    if (!currentProfile) {
+      throw new ExpressError('User not found', 404);
+    }
+    const currentProfileMedications = currentProfile.medicationList;
+    if (currentProfileMedications.length < 0) {
+      throw new ExpressError('No medication found', 404);
+    } else {
+      const currentProfileMedicationList = await Medication.findById(currentProfileMedications[0]._id);
+      res.status(StatusCodes.OK).send(currentProfileMedicationList);
```

Comment:
how about the user having more than one medication? In this case, we need to Medication.find({ _id: { $in: currentProfileMedications} });